src/rgw/rgw_data_sync.cc

```diff
@@ -4401,6 +4476,17 @@ int RGWBucketShardIncrementalSyncCR::operate()
                 return 0;
               });
       }
+
+      if (!truncated && extended_result.next_log) {
+        yield call(new RGWBucketShardIsDoneCR(sc, status_oid, bs.shard_id, extended_result, generation));
```

Comment:
hmm, it looks like this may be in the wrong place. this is inside the do-while loop, where we a) probably still have some `RGWBucketSyncSingleEntryCR`s in flight, and b) haven't written our progress to the shard status object

i think this needs to be at the very bottom of the function, where we can be sure that all `RGWBucketSyncSingleEntryCR`s succeeded, and that `marker_tracker.flush()` finished writing the shard sync status

we should also delete the shard status object after `RGWBucketShardIsDoneCR` succeeds

Reply:
as discussed deleting the shard status object will be handled in a separate pr.